### PR TITLE
logging: Cleanup and rafactoring in the internal logging macros

### PR DESCRIPTION
**Pull request by nordic-krch**
There is no functional change.
- Created macros for message filtering and use that in standard and hexdump messages (instead of having copy in both)
- Changed value used for log level (e.g. `1U` to `1`).

### CODE REVIEW

File: include/zephyr/logging/log_core.h [hunk 1]
```diff
@@ -153,6 +144,68 @@ extern "C" {
...
 	))
 
+/** @brief Compile time level checking.
+ *
+ * This check is resolved at compile time and logging message is removed if check fails.
+ *
+ * @param _level Log level.
+ * @param _source Constant data associated with the module.
+ *
+ * @retval true Message shall be complied in.
```
Comment (pdgendt) [inline include/zephyr/logging/log_core.h:154]:
```suggestion
 * @retval true Message shall be compiled in.
```

**Review by pdgendt — COMMENTED** [review]
Small typo

**Review by pdgendt — DISMISSED** [review]

**Review by pdgendt — DISMISSED** [review]

File: include/zephyr/logging/log_core.h [hunk 2]
```diff
@@ -225,69 +282,41 @@ static inline char z_log_minimal_level_to_char(int level)
...
-			  0, __VA_ARGS__); \
-	(void)_mode; \
-	if (false) { \
-		/* Arguments checker present but never evaluated.*/ \
-		/* Placed here to ensure that __VA_ARGS__ are*/ \
-		/* evaluated once when log is enabled.*/ \
-		z_log_printf_arg_checker(__VA_ARGS__); \
-	} \
-} while (false)
+#define Z_LOG2(_level, _inst, _source, ...)                                                        \
```
Comment (pdgendt) [inline include/zephyr/logging/log_core.h:291]: Missing doxygen for `_inst`

File: include/zephyr/logging/log_core.h [hunk 3]
```diff
@@ -300,70 +329,41 @@ static inline char z_log_minimal_level_to_char(int level)
...
-		break; \
-	} \
-	int mode; \
-	void *_src = IS_ENABLED(CONFIG_LOG_RUNTIME_FILTERING) ? \
-		(void *)(_dsource) : (void *)(_source); \
-	Z_LOG_MSG_CREATE(UTIL_NOT(IS_ENABLED(CONFIG_USERSPACE)), mode, \
-				  Z_LOG_LOCAL_DOMAIN_ID, _src, _level, \
-			  _data, _len, \
-			COND_CODE_0(NUM_VA_ARGS_LESS_1(_, ##__VA_ARGS__), \
+#define Z_LOG_HEXDUMP2(_level, _inst, _source, _data, _len, ...)                                   \
```
Comment (pdgendt) [inline include/zephyr/logging/log_core.h:342]: Same here

**Review by pdgendt — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**